crates/e2e/tests/e2e/colocation_hooks.rs

```diff
@@ -0,0 +1,160 @@
+use {
+    crate::setup::*,
+    ethcontract::U256,
+    model::{
+        order::{OrderCreation, OrderCreationAppData, OrderKind},
+        signature::EcdsaSigningScheme,
+    },
+    secp256k1::SecretKey,
+    serde_json::json,
+    shared::ethrpc::Web3,
+    web3::signing::SecretKeyRef,
+};
+
+#[tokio::test]
+#[ignore]
+async fn local_node_test() {
+    run_test(test).await;
+}
+
+async fn test(web3: Web3) {
+    let mut onchain = OnchainComponents::deploy(web3).await;
+
+    let [solver] = onchain.make_solvers(to_wei(1)).await;
+    let [trader] = onchain.make_accounts(to_wei(1)).await;
+    let cow = onchain
+        .deploy_cow_weth_pool(to_wei(1_000_000), to_wei(1_000), to_wei(1_000))
+        .await;
+
+    // Fund trader accounts
+    cow.fund(trader.address(), to_wei(5)).await;
+
+    // Sign a permit pre-interaction for trading.
+    let permit = cow
+        .permit(&trader, onchain.contracts().allowance, to_wei(5))
+        .await;
+    // Setup a malicious interaction for setting approvals to steal funds from
+    // the settlement contract.
+    let steal_cow = hook_for_transaction(cow.approve(trader.address(), U256::max_value()).tx).await;
+    let steal_weth = hook_for_transaction(
+        onchain
+            .contracts()
+            .weth
+            .approve(trader.address(), U256::max_value())
+            .tx,
+    )
+    .await;
+
+    tracing::info!("Starting services.");
+    let solver_endpoint = colocation::start_solver(onchain.contracts().weth.address()).await;
+    colocation::start_driver(onchain.contracts(), &solver_endpoint, &solver);
+
+    let services = Services::new(onchain.contracts()).await;
+    services.start_autopilot(vec![
+        "--account-balances-optimistic-pre-interaction-handling=true".to_string(),
+        "--enable-colocation=true".to_string(),
+        "--drivers=http://localhost:11088/test_solver".to_string(),
+    ]);
+    services
+        .start_api(vec![
+            "--enable-custom-interactions=true".to_string(),
+            "--account-balances-optimistic-pre-interaction-handling=true".to_string(),
+        ])
+        .await;
```

Comment:
This E2E test is a copy of the `hooks.rs` (former `pre_interactions.rs` E2E test) with the exception of the different setup here to use co-location instead of the legacy setup.